### PR TITLE
TCP Provider

### PR DESCRIPTION
**Pull request by fabdrol**
A simple provider to connect to a TCP NMEA0183 source (most wifi multiplexers). CC @tkurki.

### CODE REVIEW

**Comment by fabdrol** [comment]
I spliced the settings into two files and to bash files. Regarding index.js: I had to debug something so I added and removed some whitespace here and there. Functionally nothing changed.

File: settings/volare-tcp-settings.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  
+  "pipedProviders": [{
+    "id": "nmeaFromFile",
+    "pipeElements": [
+       { 
+         "type": "providers/tcp",
+         "options": {
+           "host": "192.168.0.22",
+           "port": "1337"
```
Comment (tkurki) [inline settings/volare-tcp-settings.json]: You can add here optionMappings to make the host & port configurable/overridable from the command line as well.

**Comment by tkurki** [comment]
I was thinking about the same use case (tcp server not started but Signal K already up). Things get a also booted/loose power.

Have you looked at https://github.com/juliangruber/reconnect-core or https://github.com/foreverjs/nssocket for out of the box reconnection logic?

Primus apparently also has tcp available: "Now we are able to connect to a vanilla tcp server (Java?) and still keep all the neat abstractions (reconnect, heartbeat, etc) that Primus provides." https://www.npmjs.com/package/primus-tcp

**Comment by fabdrol** [comment]
@tkurki I actually missed your comments re the reconnect stuff, but it's quite trivial so I don't see the need to depend on something external for that. I've just pushed my latest changes; had some weird git problems so I might have to do something about that tomorrow. 

It's not quite ready yet for merging -- I need to test with an actual TCP source as I had some issues with my (quick'n'dirty) TCP "simulator" (transmits a log file line-by-line in about 10 lines of code.. so not very solid). Mostly timeout issues, so I reckon connecting to an actual source might improve things. I'll do that this weekend on the boat.

**Comment by fabdrol** [comment]
I've been testing this on the boat last weekend, I don't have the timeouts with an actual TCP source but it could use some extra testing. @tkurki could you do some additional testing after I've cleaned up the code a bit (removed the testing code, integrated it in the setting file, etc)?

**Comment by tkurki** [comment]
Should this be merged now?

**Comment by fabdrol** [comment]
@tkurki shall we merge this?

**Pull request merged**